_commands/basics/echo.md

```diff
@@ -3,22 +3,71 @@
 
 echo
 ---
-TODO: Add documentation for this command by submitting a pull request.
+
+`echo` is used to display message on screen, and write given string to standard output, with each a space between each and a newline after the last one.  `echo` is implemented as builtin command in many command shells like bash, ksh, csh.
 <!-- one line explanation would go here -->
 
 <!-- minimal example -->
 ~~~ bash
-*short* example of command usage and output
+$ echo "Hello World"
+Hello World
+$ echo Hello World
+Hello World
 ~~~
 
 <!--more-->
 
+
 ### Useful Options / Examples
 
 #### Example command
+`echo` output can be redirected into a file for easy use. 
+~~~ bash
+$ echo Hello World > example.txt
+$ cat example.txt
+Hello World
+~~~
+#### Example command
+`echo` output can also be appended to a file.
+~~~ bash
+$ echo Hello > example.txt
+$ echo This is a new line. >> example.txt
+$ cat example.txt 
+Hello
+This is a new line.
+~~~
+#### Example command
+##### `echo -n` 
+`-n` is a usefule flag if you don't want the trailing newline after the string
 
-##### Break it down
+~~~ bash
+$ echo -n "Hello World." > example.txt
```

Comment:
Instead of redirecting the output to a file, could we show how the output looks on the terminal? That way we don't have to invoke the `cat` command and make things a bit more confusing by using two commands instead of just `echo.